dask/tests/test_optimize.py

```diff
@@ -28,90 +28,220 @@ def test_cull():
+        'a': (inc, 'x'),
+        'b': (inc, 'x'),
+        'c': (inc, 'x'),
+        'd': (inc, 'c'),
+        'x': (inc, 'y'),
+        'y': 0,
+    }
+    dsk, dependencies = fuse(d, rename_fused_keys=False)
+    assert dsk == {
+        'a': (inc, 'x'),
+        'b': (inc, 'x'),
+        'd': (inc, (inc, 'x')),
+        'x': (inc, 0)
+    }
+    assert dependencies == {'x': set(), 'd': set(['x']),
+                            'a': set(['x']), 'b': set(['x'])}
+
+    dsk, dependencies = fuse(d, rename_fused_keys=True)
+    assert dsk == {
+        'a': (inc, 'y-x'),
+        'b': (inc, 'y-x'),
+        'c-d': (inc, (inc, 'y-x')),
+        'y-x': (inc, 0),
+        'd': 'c-d',
+        'x': 'y-x',
+    }
+    assert dependencies == {'y-x': set(), 'c-d': set(['y-x']),
+                            'a': set(['y-x']), 'b': set(['y-x']),
+                            'd': set(['c-d']), 'x': set(['y-x'])}
+
+    d = {
+        'a': 1,
+        'b': (inc, 'a'),
+        'c': (add, 'b', 'b'),
+    }
+    dsk, dependencies = fuse(d, rename_fused_keys=False)
+    assert dsk == {
+        'b': (inc, 1),
+        'c': (add, 'b', 'b'),
+    }
+    assert dependencies == {'b': set(), 'c': set(['b'])}
+
+    dsk, dependencies = fuse(d, rename_fused_keys=True)
+    assert dsk == {
+        'a-b': (inc, 1),
+        'c': (add, 'a-b', 'a-b'),
+        'b': 'a-b',
+    }
+    assert dependencies == {'a-b': set(), 'c': set(['a-b']), 'b': set(['a-b'])}
+
 
-    assert dsk == {'w': (inc, 'x'),
-                   'x': (inc, (inc, 'z')),
-                   'z': (add, 'a', 'b'),
-                   'a': 1,
-                   'b': 2 }
+def test_fuse_keys():
+    d = {
+        'a': 1,
+        'b': (inc, 'a'),
+        'c': (inc, 'b'),
+    }
+    keys = ['b']
+    dsk, dependencies = fuse(d, keys, rename_fused_keys=False)
+    assert dsk == {
+        'b': (inc, 1),
+        'c': (inc, 'b'),
+    }
+    assert dependencies == {'b': set(), 'c': set(['b'])}
+
+    dsk, dependencies = fuse(d, keys, rename_fused_keys=True)
+    assert dsk == {
+        'a-b': (inc, 1),
+        'c': (inc, 'a-b'),
+        'b': 'a-b',
+    }
+    assert dependencies == {'a-b': set(), 'c': set(['a-b']), 'b': set(['a-b'])}
+
+    d = {
+        'w': (inc, 'x'),
+        'x': (inc, 'y'),
+        'y': (inc, 'z'),
+        'z': (add, 'a', 'b'),
+        'a': 1,
+        'b': 2,
+    }
+    keys = ['x', 'z']
+    dsk, dependencies = fuse(d, keys, rename_fused_keys=False)
+    assert dsk == {
+        'w': (inc, 'x'),
+        'x': (inc, (inc, 'z')),
+        'z': (add, 'a', 'b'),
+        'a': 1,
+        'b': 2 ,
+    }
     assert dependencies == {'a': set(), 'b': set(), 'z': set(['a', 'b']),
                             'x': set(['z']), 'w': set(['x'])}
 
+    dsk, dependencies = fuse(d, keys, rename_fused_keys=True)
+    assert dsk == {
+        'w': (inc, 'y-x'),
+        'y-x': (inc, (inc, 'z')),
+        'z': (add, 'a', 'b'),
+        'a': 1,
+        'b': 2 ,
+        'x': 'y-x',
```

Comment:
I can't answer this, but I'll comment anyway.  Keeping `{old_key: new_key}` aliases in the fused dict isn't something I was originally expecting to need to do.  It makes sense, though, since `keys` in `get(d, keys)` may refer to the original keys that were known at construction time and expected to exist.  It's possible we can be smarter about which aliases to keep by using `keys=` argument in `fuse` (there were so many failing tests without including aliases that I was too afraid to explore this option initially).

Note that no tasks in the fused dask dict refer to the old keys that are now aliases.